src/DIRAC/WorkloadManagementSystem/JobWrapper/Watchdog.py

```diff
@@ -287,12 +281,14 @@ def _performChecks(self):
             self.parameters["LoadAverage"] = []
         self.parameters["LoadAverage"].append(loadAvg)
 
-        memoryUsed = self.getMemoryUsed()
-        msg += f"MemUsed: {memoryUsed:.1f} kb "
-        heartBeatDict["MemoryUsed"] = memoryUsed
+        result = self.profiler.memoryUsage(withChildren=True)
+        if not result["OK"]:
+            self.log.warn("Could not get rss info from profiler", result["Message"])
+        msg += f"MemUsed: {result['Value']:.1f} MB "
```

Comment:
I don't see anything in the [profiler](https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/Core/Utilities/Profiler.py) that would add "Value" if S_ERROR is returned, which means this line is an exception if result is not OK? Or am I missing something?

Reply:
The `checkInvocation` decorator can return `S_ERROR`